Automattic/wp-calypso
# ci(e2e): Refactor screenrecorder and framebuffer logs

#### Changes proposed in this Pull Request

* Split `video-recorder.js` into two set of hooks: framebuffer and video recorder, as they are separate concerns.
* Drop support for `neverSaveScreenshots` and `saveAllScreenshots`. They are not documented anywhere, and from what I can see, never used.
* Drop support for adding the URL as a png tag. You couldn't easily read them back without specialized tools anyway.
* Clean up the logic to toggle framebuffer and video recording support. Now it is centralized in one single file instead of every method checking if video was available.
* Drop slack notifications when a screenshot couldn't be saved.
* Save videos and screenshots in the same directory for easier asset management
* Add hook to close the browser. Root Hooks run after the hooks declared in `./mocha-hooks.js`. Therefore, closing the browser must be the last `afterAll` hook in Root Hooks, in case previous `afterAll` hooks want to use it (eg: to take a screenshot)

#### Testing instructions

Verify e2e build still saves screenshots and video recordings.

Example:
![image](https://user-images.githubusercontent.com/975703/118133732-74a6fd80-b401-11eb-8200-21b9d301ce62.png)
